deltacat/storage/model/metafile.py

```diff
@@ -22,13 +27,153 @@
+                "You can also run AWS CLI command to get more detailed error message "
+                "(e.g., aws s3 ls <file-name>). "
+                "See https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3/index.html "  # noqa
+                "and https://docs.ray.io/en/latest/data/creating-datasets.html#reading-from-remote-storage "  # noqa
+                "for more information."
+            )
+        )
+    else:
+        raise error
+
+
+def _expand_directory(
+    path: str,
+    filesystem: pyarrow.fs.FileSystem,
+    exclude_prefixes: Optional[List[str]] = None,
+    ignore_missing_path: bool = False,
+) -> List[Tuple[str, int]]:
+    """
+    Expand the provided directory path to a list of file paths.
+
+    Args:
+        path: The directory path to expand.
+        filesystem: The filesystem implementation that should be used for
+            reading these files.
+        exclude_prefixes: The file relative path prefixes that should be
+            excluded from the returned file set. Default excluded prefixes are
+            "." and "_".
+
+    Returns:
+        An iterator of (file_path, file_size) tuples.
+    """
+    if exclude_prefixes is None:
+        exclude_prefixes = [".", "_"]
+
+    from pyarrow.fs import FileSelector
+
+    selector = FileSelector(path, recursive=True, allow_not_found=ignore_missing_path)
+    files = filesystem.get_file_info(selector)
+    base_path = selector.base_dir
+    out = []
+    for file_ in files:
+        if not file_.is_file:
+            continue
+        file_path = file_.path
+        if not file_path.startswith(base_path):
+            continue
+        relative = file_path[len(base_path) :]
+        if any(relative.startswith(prefix) for prefix in exclude_prefixes):
+            continue
+        out.append((file_path, file_.size))
+    # We sort the paths to guarantee a stable order.
+    return sorted(out)
+
+
+def _get_file_infos(
+    path: str,
+    filesystem: pyarrow.fs.FileSystem,
+    ignore_missing_path: bool = False,
+) -> List[Tuple[str, int]]:
+    """Get the file info for all files at or under the provided path."""
+    file_infos = []
+    try:
+        file_info = filesystem.get_file_info(path)
+    except OSError as e:
+        _handle_read_os_error(e, path)
+    if file_info.type == FileType.Directory:
+        for (file_path, file_size) in _expand_directory(path, filesystem):
+            file_infos.append((file_path, file_size))
+    elif file_info.type == FileType.File:
+        file_infos.append((path, file_info.size))
+    elif file_info.type == FileType.NotFound and ignore_missing_path:
+        pass
+    else:
+        raise FileNotFoundError(path)
+
+    return file_infos
+
+
+def _get_file_info(
+    path: str,
+    filesystem: pyarrow.fs.FileSystem,
+    ignore_missing_path: bool = False,
+) -> Union[FileInfo, List[FileInfo]]:
+    """Get the file info or list of file infos for the provided path."""
```

Comment:
Would this ever return a list of file infos (if only a single `path` str is provided)?

Reply:
Good catch - this was left behind during refactoring out of `_get_file_infos()`, and _shouldn't_ ever return a list now. 
```suggestion
    """Get the file info for the provided path."""
```